inc/dashboard/namespace.php

```diff
@@ -0,0 +1,436 @@
+	$sort_order = 'SORT_' . strtoupper( $order );
+
+	array_multisort( $avg_conversion_rate, $sort_order, $list );
+
+	return $list;
+}
+
+/**
+ * Get data from a single block.
+ *
+ * @uses get_views_List()
+ *
+ * @param string $block_id Get a single block's analytics data pulled from the ES query.
+ *
+ * @return array The block's analytics data.
+ */
+function get_block_data( string $block_id ) : array {
+	$data = get_views_list();
+	return $data[ $block_id ];
+}
+
+/**
+ * Get analytics views list data.
+ *
+ * @todo Integrate date range searches.
+ *
+ * @param string $order How to order the data. Accepted values are 'asc' and 'desc'. Defaults to 'desc'.
+ * @param int $days How many days worth of analytics to fetch. Defaults to 7.
+ */
+function get_views_list( string $order = 'desc', int $days = 7 ) : array {
+	$query = [
+		'query' => [
+			'bool' => [
+				'filter' => [
+					[
+						// Query from the current site.
+						'term' => [
+							'attributes.blogId.keyword' => get_current_blog_id(),
+						],
+					],
+					[
+						// We're just interested in views.
+						'terms' => [
+							'event_type.keyword' => [
+								'experienceView',
+								'conversion',
+							],
+						],
+					],
+				],
+			],
+		],
+		'aggs' => [
+			'blocks' => [
+				'terms' => [
+					// Get the block data. This will give us the key for the block, which is stored as the post slug.
+					'field' => 'attributes.clientId.keyword',
+					'size' => 10000, // Use arbitrary large size that is more than we're likely to need.
+					'order' => [
+						'_count' => $order,
+					],
+				],
+				'aggs' => [
+					'events' => [
+						'terms' => [
+							'field' => 'event_type.keyword',
+						],
+					],
+				],
+			],
+		],
+		'size' => 0,
+	];
+
+	// 1: Sort order, asc or desc.
+	// 2: Start date to query by.
+	// 3: End date to query by.
+	$key = sprintf( 'views:list:%1$s:days:%2$d', $order, $days );
+	$cache = wp_cache_get( $key, 'altis-xbs' );
+
+	if ( $cache ) {
+		return $cache;
+	}
+
+	$result = query( $query );
+
+	// Check for a result before getting aggregated data.
+	$data = ( $result ) ? get_aggregate_data( $result['aggregations']['blocks']['buckets'] ?? [] ) : [];
+
+	// Don't cache anything if we didnt' get a result.
+	if ( ! empty( $data ) ) {
+		wp_cache_set( $key, $data, 'altis-xbs', 5 * MINUTE_IN_SECONDS );
+	}
+
+	return $data;
+}
+
+/**
+ * Alter the default WP_Query to change the sort order.
+ *
+ * @todo Finish this function.
+ *
+ * @param \WP_Query $query The WP_Query object.
+ *
+ * @return \WP_Query The updated WP_Query object.
+ */
+function modify_views_list_query( $query ) {
+	if (
+		// Bail if we arent' in the admin.
+		! is_admin() ||
+		// Bail if we aren't looking at the XB Insights page.
+		$query->get( 'post_type' ) !== 'xb'
+	) {
+		return $query;
+	}
+
+	$order = $query->get( 'order' ) ?: 'desc';
+	$orderby = $query->get( 'orderby' ) ?: 'views';
+	$days = get_days_view();
```

Comment:
Might be good to support this via a query arg too, eg. `$days = $query->get( 'days' ) ?: get_days_view();`